test/linalg/cholesky.jl

```diff
@@ -217,15 +217,17 @@ end
     A = complex.(randn(10,5), randn(10, 5))
     v = complex.(randn(5), randn(5))
     for uplo in (:U, :L)
-        AcA = A'A
-        BcB = AcA + v*v'
-        BcB = (BcB + BcB')/2
-        F = cholfact(Hermitian(AcA, uplo))
-        G = cholfact(Hermitian(BcB, uplo))
-        @test LinAlg.lowrankupdate(F, v)[uplo] ≈ G[uplo]
-        @test_throws DimensionMismatch LinAlg.lowrankupdate(F, ones(eltype(v), length(v)+1))
-        @test LinAlg.lowrankdowndate(G, v)[uplo] ≈ F[uplo]
-        @test_throws DimensionMismatch LinAlg.lowrankdowndate(G, ones(eltype(v), length(v)+1))
+        @eval begin
+            AcA = $(A)'*$(A)
+            BcB = AcA + $v*$v'
+            BcB = (BcB + BcB')/2
+            F = cholfact(Hermitian(AcA, $(QuoteNode(uplo))))
+            G = cholfact(Hermitian(BcB, $(QuoteNode(uplo))))
+            @test LinAlg.lowrankupdate(F, $v).$uplo ≈ G.$uplo
+            @test_throws DimensionMismatch LinAlg.lowrankupdate(F, ones(eltype($v), length($v)+1))
+            @test LinAlg.lowrankdowndate(G, $v).$uplo ≈ F.$uplo
+            @test_throws DimensionMismatch LinAlg.lowrankdowndate(G, ones(eltype($v), length($v)+1))
+        end
```

Comment:
Perhaps explicit `getproperty` calls (if possible?) would be simpler than metafication?

Reply:
The idea was to exercise the compile time version but that is probably not worth the metafication here.